src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,178 @@
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvException;
+
+import seedu.address.commons.exceptions.DataLoadingException;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.AddCommandParser;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+
+/**
+ * Changes the  of an existing person in the address book.
+ */
+public class ImportCommand extends Command {
+
+    public static final String COMMAND_WORD = "import";
+
+    public static final String MESSAGE_NOT_IMPLEMENTED_YET =
+            "Remark command not implemented yet";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports contacts from specified filepath "
+            + "Existing contacts will.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_IMPORT + "import] "
+            + "Example: " + COMMAND_WORD + PREFIX_IMPORT + "C:usr/lib/text.csv";
+    private static final String MESSAGE_ARGUMENTS = "filePath: %s";
+    private static final String MESSAGE_IMPORT_SUCCESS = "Imported Contacts from: %s";
+    private static final String MESSAGE_DATA_LOAD_ERROR = "Unable to load data from %s";
+    private static final String MESSAGE_PARSE_ERROR = "Invalid data format in %s";
+    private final Path filePath;
+    private final AddCommandParser addCommandParser = new AddCommandParser();
+
+    /**
+     * Represents the order of the data that should be parsed into the addCommandParser
+     */
+    private final String[] header = {"name", "phone", "email", "address", "matric", "reflection", "studio", "tags"};
+
+    /**
+     * Represents a mapping of String to prefix of the data that should be parsed into the addCommandParser.
+     */
+    private final Map<String, Prefix> prefixMap = Map.of(
+            "name", PREFIX_NAME,
+            "phone", PREFIX_PHONE,
+            "email", PREFIX_EMAIL,
+            "address", PREFIX_ADDRESS,
+            "matric", PREFIX_MATRIC_NUMBER,
+            "reflection", PREFIX_REFLECTION,
+            "studio", PREFIX_STUDIO,
+            "tags", PREFIX_TAG
+    );
+
+    /**
+     * @param filePath absolute path of file (path starts from C:...)
+     */
+    public ImportCommand(Path filePath) {
+        requireNonNull(filePath);
+
+        this.filePath = filePath;
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        try {
+            List<Map<String, String>> data = readCsvFile();
+            for (Map<String, String> personData : data) {
+                try {
+                    String addCommandInput = convertToAddCommandInput(personData);
```

Comment:
great job using helper functions to abstract some of the functionality away from this method to make the execute method easier to understand!

Reply:
Still needs more slap tho